packages/react-router/src/ScrollRestoration.tsx

```diff
@@ -0,0 +1,54 @@
+import { useRouter } from './useRouter'
+import {
+  defaultGetScrollRestorationKey,
+  getCssSelector,
+  scrollRestorationCache,
+} from './scroll-restoration'
+import type { ScrollRestorationOptions } from './scroll-restoration'
+import type { ParsedLocation } from '@tanstack/router-core'
+
+/**
+ * @deprecated use createRouter's `scrollRestoration` option instead
+ */
+export function ScrollRestoration(_props: ScrollRestorationOptions) {
```

Comment:
Is this considered a breaking change? Where do we draw the line on this one?

Like builds won't fail since the component is still being exported... But at the same-time, any existing scroll restoration functionality will break on the user/app facing experience.